tests/std/tests/VSO_0180469_ptr_cat/test.cpp

```diff
@@ -377,10 +388,13 @@ void equal_safe_test_cases() {
     test_case_Equal_memcmp_is_safe<sizeof(int) == sizeof(long), unsigned long, unsigned int>();
     test_case_Equal_memcmp_is_safe<true, long long, unsigned long long>();
     test_case_Equal_memcmp_is_safe<true, unsigned long long, long long>();
-    // memcmp is not safe for bool types
-    test_case_Equal_memcmp_is_safe<false, bool, bool>();
-    test_case_Equal_memcmp_is_safe<false, bool, char>();
-    test_case_Equal_memcmp_is_safe<false, char, bool>();
+#pragma warning(push)
+#pragma warning(disable : 4806) // no value of type '_Elem1' promoted to type '_Elem2' can equal the given constant

```

Comment:
It appears that this warning is being emitted by the `_Can_memcmp_elements` machinery - so this push/disable/pop should appear in `<xutility>` rather than in "user" code. (We suppress warnings in tests when the tests themselves are doing something strange-but-okay. Here, calling `std::equal` on ranges of `bool` and `char` is perfectly fine, so the product code needs to suppress the warning from internal machinery.)

Reply:
I went ahead and pushed a change for this.